### ISSUE TITLE
if attribute on <property> is not working

### ISSUE THREAD

**Issue by FabienPoirier** [nant/nant#110]
Hi, 

I am doing NANT for years and I like it very much. But never I could make work attribute if on property task, as is it documented.

It could be nice it it were working has documented. 

Regards
Fabien

**Comment by rmboggs** [comment]
I take it back, I am able to get an error if I try to do something like this:
```xml
<?xml version="1.0"?>
<project name="proptest">
    <echo message="Test 1 = '${test.one}'" if="${property::exists('test.one')}"/>
    <echo message="This should be the only message displayed"/>
</project>
```
The output is this:
```
NAnt 0.93 (Build 0.93.5470.0; nightly; 12/23/2014)
Copyright (C) 2001-2014 Gerry Shaw
http://nant.sourceforge.net

Buildfile: file:///C:/temp/PropertyTest.xml
Target framework: Microsoft .NET Framework 4.0


BUILD FAILED

C:\temp\PropertyTest.xml(3,6):
Property evaluation failed.
Expression: Test 1 = '${test.one}'
                        ^^^^^^^^
    Property 'test.one' has not been set.

Total time: 0 seconds.
```
Even though I am using the if attribute to test to see if the property exists before I echo it, NAnt is still trying to expand the non-existent var.  I have a fix for this which I am trying to fine tune (hopefully over the weekend) which may also solve for #108 as well.  I'll create another pull request for review once it is ready.

**Comment by rmboggs** [comment]
Just merged fix.  Thanks for the report.

**Comment by FabienPoirier** [comment]
Hi,

In fact the issue was bad, it is occurring when

<echo if="${property::exists('LNG.file')}" message="LNG.file : ${LNG.file}"/>

If property is not set, there is an error. Sorry for the delay, but I just get it now. May be I am wrong using it.

Cordialement

[logo 3]


Fabien POIRIER
Directeur Technique
INFORMIA
Ch. Saint Lambert, Actiparc II, Bât. A
13821 La Penne sur Huveaune
Tél. :        +33 (0)4.91.27.10.37
f.poirier@informia.fr<mailto:f.poirier@informia.fr>
www.informia.fr<http://www.informia.fr/>
[callme]<skype:fabien.poirier.informia?call>






De : Ryan Boggs [mailto:notifications@github.com]
Envoyé : mercredi 31 décembre 2014 19:16
À : nant/nant
Cc : Fabien Poirier
Objet : Re: [nant] if attribute on <property> is not working (#110)


I cannot seem to recreate this issue. The script I am testing is below:

<?xml version="1.0"?>

<project name="proptest">

    <property name="test.one" value="first value"/>

    <property name="test.one" value="second value" if="${property::exists('does.not.exist')}"/>

    <echo message="Test 1 = '${test.one}'"/>



    <property name="test.two" value="should not exist" if="${property::exists('does.not.exist')}"/>

    <if test="${property::exists('test.two')}">

        <echo message="Test 2 = '${test.two}'"/>

    </if>

    <if test="${not property::exists('test.two')}">

        <echo message="Test 2 does not exist"/>

    </if>

</project>

The output I get is a follows:

NAnt 0.93 (Build 0.93.5470.0; nightly; 12/23/2014)

Copyright (C) 2001-2014 Gerry Shaw

http://nant.sourceforge.net



Buildfile: file:///C:/absolute/path/to/nant-0.93-nightly-2014-12-23/bin/PropertyTest.xml<file:///C:\absolute\path\to\nant-0.93-nightly-2014-12-23\bin\PropertyTest.xml>

Target framework: Microsoft .NET Framework 4.0



     [echo] Test 1 = 'first value'

     [echo] Test 2 does not exist



BUILD SUCCEEDED



Total time: 0 seconds.

Is there something I am missing?

**Comment by rmboggs** [comment]
Check the latest nightly build.  I believe that I fixed this some time ago. Let me know if this is still an issue so I can reopen.
